src/player.cpp

```diff
@@ -3712,6 +3712,9 @@ void player::practice( const skill_id &id, int amount, int cap, bool suppress_wa
         get_skill_level_object( id ).train( amount );
         int newLevel = get_skill_level( id );
         std::string skill_name = skill.name();
+        if( newLevel > oldLevel ) {
+            g->events().send<event_type::gains_skill_level>( getID(), id, newLevel );
```

Comment:
Why this extra if clause? Why not adding this in to the one with "Your skill in ... has increased"?
Achievements can only be done by the players character, as far as I'm aware, so that would make sense...

Reply:
Achievements do check for player being player, but that's being done on json level and technically can be omitted, so I'm not taking any chances. That might sound like an overkill but I'd rather have it this way, then having someone in the future scratching their head why some non-player event didn't trigger.